storage/gc_queue_test.go

```diff
@@ -314,6 +314,138 @@ func TestGCQueueProcess(t *testing.T) {
 	}
 }
 
+func TestGCQueueTransactionTable(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	const now time.Duration = 3 * 24 * time.Hour
+	const dAbandon = -2 * DefaultHeartbeatInterval
+	type spec struct {
+		status      roachpb.TransactionStatus
+		ts          time.Duration
+		heartbeatTS time.Duration
+		intentSpans []roachpb.Span
+		seqGC       bool // expect sequence cache entries removed?
+	}
+	// Describes the state of the Txn table before the test.
+	before := map[string]spec{
+		// Too young, should not touch.
+		"a": {roachpb.PENDING, now - txnCleanupThreshold + 1, 0, []roachpb.Span{{Key: roachpb.Key("q")}}, false},
+		// Old, but still heartbeat. No GC.
+		"b": {roachpb.PENDING, 0, now - txnCleanupThreshold + 1, nil, false},
+		// Old and aborted, should delete.
+		"c": {roachpb.ABORTED, now - txnCleanupThreshold - 1, 0, nil, true},
+		// Abandoned and pending, so should push and abort it successfully.
+		// But it's not old enough to actively go after the sequence cache.
+		// There's some room for optimization here by ways of sharing the
+		// transaction grooming results with the corresponding operation for
+		// the sequence cache, but that's only worthwhile in weird cases.
+		"d": {roachpb.PENDING, now - dAbandon, 0, nil, false},
```

Comment:
`dAbandon` is negative, so `now - dAbandon` is in the future, which doesn't seem to be what is intended here.